pkg/xds/topology/dataplanes.go

```diff
@@ -3,23 +3,22 @@ package topology
 import (
 	"context"
 
-	"github.com/kumahq/kuma/pkg/core/dns"
+	"github.com/kumahq/kuma/pkg/core/runtime"
 
 	"github.com/pkg/errors"
 
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
-	"github.com/kumahq/kuma/pkg/core/resources/manager"
 )
 
 // GetDataplanes returns list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
-func GetDataplanes(ctx context.Context, rm manager.ReadOnlyResourceManager, mesh string) (*core_mesh.DataplaneResourceList, error) {
+func GetDataplanes(rt runtime.RuntimeContext, ctx context.Context, mesh string) (*core_mesh.DataplaneResourceList, error) {
```

Comment:
nit: I'd rather have GetDataplanes to depend on `ReadOnlyResourceManager` and `LookupIP`. Dependencies of the functions are then clear. Otherwise you need to figure out what exactly is picked up from RuntimeContext. Also it should be easier to test then